### PR TITLE
deploy/jsonnet: Make PSP optional

### PR DESCRIPTION
**Pull request by kakkoyun**
Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

Make PSP optional and disable it by default `Warning: policy/v1beta1 PodSecurityPolicy is deprecated in v1.21+, unavailable in v1.25+`

For PSP:
- I have followed https://kubernetes.io/docs/tasks/configure-pod-container/migrate-from-psp/ as much as possible.
- Also adds a namespace with admission control for development: https://kubernetes.io/docs/tutorials/security/cluster-level-pss/
- https://jamesdefabia.github.io/docs/user-guide/security-context/

For Capabilites:
- https://man7.org/linux/man-pages/man7/capabilities.7.html
- https://docs.docker.com/engine/reference/run/#runtime-privilege-and-linux-capabilities

ref: https://github.com/parca-dev/parca/pull/3142

### CODE REVIEW

File: deploy/lib/parca-agent/parca-agent.libsonnet [hunk 1]
```diff
@@ -283,7 +193,14 @@ function(params) {
           '--metadata-disable-caching',
         ] else []
       ),
-      securityContext: pa.config.securityContext,
+      // Container level security context.
+      securityContext: {
+        privileged: true,
+        allowPrivilegeEscalation: true,
+        capabilities: {
+          add: ['BPF', 'PERFMON'],  // SYS_ADMIN
```
Comment (brancz) [inline deploy/lib/parca-agent/parca-agent.libsonnet:201]: Can we maybe first test this on our demo and prod cluster?

File: deploy/lib/parca-agent/parca-agent.libsonnet [hunk 2]
```diff
@@ -465,4 +383,96 @@ function(params) {
       },
     },
   },
+
+  [if std.objectHas(params, 'podSecurityPolicy') && params.podSecurityPolicy then 'podSecurityPolicy']: {
```
Comment (brancz) [inline deploy/lib/parca-agent/parca-agent.libsonnet:387]: love this!

**Review by brancz — APPROVED** [review]

**Comment by brancz** [comment]
Woah how cool is it that the e2e test failed? Amazing!

File: deploy/lib/parca-agent/parca-agent.libsonnet (hunk 1 above)
Comment (kakkoyun) [reply deploy/lib/parca-agent/parca-agent.libsonnet:203]: 👍 I will do it. I have tested locally against minikube it worked.

**Comment by kakkoyun** [comment]
> Woah how cool is it that the e2e test failed? Amazing!

It should be fixed now :)

**Review by metalmatze — APPROVED** [review]

File: deploy/lib/parca-agent/parca-agent.libsonnet (hunk 1 above)
Comment (kakkoyun) [reply deploy/lib/parca-agent/parca-agent.libsonnet]: This was a good call. In theory, this should work, but this made e2e tests fail in practice.
So we will stick with `SYS_ADMIN` for now. It requires more investigation, which additional capabilities we need besides `BPF` and `PERFMON`

Comment (v-thakkar) [reply deploy/lib/parca-agent/parca-agent.libsonnet]: Maybe kernel version issue? I think CAP_BPF is only available after 5.8.

**Pull request merged**